hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/pipeline/Pipeline.java

```diff
@@ -301,14 +306,21 @@ public static Pipeline getFromProtobuf(HddsProtos.Pipeline pipeline)
     for (DatanodeDetailsProto member : pipeline.getMembersList()) {
       nodes.add(DatanodeDetails.getFromProtoBuf(member));
     }
+    UUID leaderId = null;
+    if (pipeline.hasLeaderID() &&
+        StringUtils.isNotEmpty(pipeline.getLeaderID())) {
+      leaderId = UUID.fromString(pipeline.getLeaderID());
```

Comment:
Correct me if I'm wrong, with current approach, the improvement will be limited as fromProtoBuf to String will prefer process UUID before ID128. And to protoBuf will need to add both String version of UUID and the long/long version of uuid 128.

Reply:
@xiaoyuyao  Sorry for missing the comment. @elek  Thanks for reminding.

> And to protoBuf will need to add both String version of UUID and the long/long version of uuid 128
This is for compatibility, in the future, I will remove String version of UUID.

> The improvement will be limited as fromProtoBuf to String will prefer process UUID before ID128
Do you mean we should prefer ID128 before process UUID.fromString ? If so, I think you are totally right, I will update it.  Besides I will also remove  UUID.fromString in the future.

Reply:
@xiaoyuyao Thanks for review. I have updated the patch, Could you help review it again ?